packages/zowe-explorer/src/shared/actions.ts

```diff
@@ -236,3 +236,20 @@ export async function resetValidationSettings(node: IZoweNodeType, setting: bool
     }
     return node;
 }
+
+/**
+ * Reports the progress of a Gui.withProgress action results in visual update for users.
+ *
+ * @export
+ * @param {any} progress - Gui.withProgress progress
+ * @param {number} valueLength - values array length used to calculate progress
+ * @param {number} index - withProgress loop array's index
+ * @param {string} action - the action the progress is reported for, ie. Uploading
+ */
+export function reportProgress(progress: any, valueLength: number, index: number, action: string): void {
```

Comment:
If this method can be used with any `Gui.withProgress` action, would it make sense to define it in the `Gui` namespace of the ZE API package?